Modules/Sources/NetworkingCore/Network/AlamofireNetworkErrorHandler.swift

```diff
@@ -233,11 +239,16 @@ private extension AlamofireNetworkErrorHandler {
     }
 
     func clearUnsupportedFlag(for siteID: Int64) {
-        queue.sync(flags: .barrier) {
+        // Use dedicated serial queue for UserDefaults operations to:
+        // 1. Prevent race conditions where concurrent writes overwrite each other
+        // 2. Avoid deadlock by not using the main queue that KVO observers may need
+        userDefaultsQueue.async { [weak self] in
```

Comment:
Same concerc as above, `clearUnsupportedFlag` might return befire the value is set in UserDefault